### ISSUE TITLE
Trying to use "expires_at" fails with error

### ISSUE THREAD

**Issue by Pigueiras** [egnyte/gitlabform#207]
**Describe the bug**

With something like:

```
project_settings
  myproject:
    members:
      users:
        user:
          access_level: 30
          expires_at: 2021-05-12
```

I get:

```
Traceback (most recent call last):
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlabform/core.py", line 384, in process_all
    self.project_processors.process_project(
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlabform/processors/project/__init__.py", line 63, in process_project
    processor.process(project_and_group, configuration, dry_run, output_file)
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlabform/processors/util/decorators.py", line 42, in method_wrapper
    return method(self, project_and_group, SafeDict(configuration), *args)
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlabform/processors/abstract_processor.py", line 47, in process
    self._process_configuration(project_or_project_and_group, configuration)
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlabform/processors/project/members_processor.py", line 44, in _process_configuration
    self.gitlab.add_member_to_project(
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlab/members.py", line 12, in add_member_to_project
    return self._make_requests_to_api(
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlab/core.py", line 137, in _make_requests_to_api
    response = self._make_request_to_api(
  File "/Users/pigueira/python_stuff/lib/python3.8/site-packages/gitlabform/gitlab/core.py", line 193, in _make_request_to_api
    json.dumps(dict_data, sort_keys=True),
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type date is not JSON serializable
```

It looks like the yaml loader gets the `expires_at` variable as a `datetime` but the json serializer doesn't know what to do with datetimes.

```
import json
import yaml

read = yaml.safe_load('''
a: 2024-03-03
''')

json.dumps(read)  # kaboom
```

I couldn't find any test with `expires_at` so I guess that's why it was never discovered. A simple fix would be to add strftime("%Y-%m-%d") there:

https://github.com/egnyte/gitlabform/blob/c7ab9a0d59bf16d3716aa05a009ac13c8e5d5caf/gitlabform/gitlabform/processors/project/members_processor.py#L41

**GitLabForm version**

GitLabForm version: 1.23.0 (the latest)

**Comment by Pigueiras** [comment]
I discovered this implementing the fix for #101 @gdubicki , do you want a separate PR for this or is it OK if it's all together?

**Comment by gdubicki** [comment]
Thanks for reporting @Pigueiras ! You can fix it in that PR, no problem.

**Comment by bbpadministrator** [comment]
This is fixed by #208

**Comment by Pigueiras** [comment]
This is fixed by #208
